rs-matter-macros-impl/src/idl.rs

```diff
@@ -685,4 +707,100 @@ mod tests {
             )
         );
     }
+
+    #[test]
+    fn struct_fields_string() {
+        let idl = parse_idl(
+            "
+              cluster TestForStructs = 1 {
+                struct WithStringMember {
+                  char_string<16> short_string = 1;
+                  long_char_string<512> long_string = 2;
+                  optional char_string<32> opt_str = 3;
+                  optional nullable long_char_string<512> opt_nul_str = 4;
+                }
+              }
+            ",
+        );
+
+        let cluster = get_cluster_named(&idl, "TestForStructs").expect("Cluster exists");
+        let context = IdlGenerateContext::new("rs_matter_crate");
+
+        let defs: TokenStream = cluster
+            .structs
+            .iter()
+            .map(|c| struct_definition(c, &context))
+            .collect();
+
+        assert_tokenstreams_eq!(
+            &defs,
+            &quote!(
+                #[derive(
+                    Debug,
+                    PartialEq,
+                    Eq,
+                    Clone,
+                    Hash,
```

Comment:
Just a note: apparently I was wrong yesterday, and `Hash` does not need neither STD, nor `alloc`!
`std::hash::Hash` is just an alias for `core::hash::Hash`.

Sorry for yesterday's confusion!